modules/contributor-guide/partials/proc_adding-a-che-theia-plug-in-into-a-workspace.adoc

```diff
@@ -7,67 +7,64 @@
-image::extensibility/adding-plug-ins-in-workspace-devfile.png[link="../_images/extensibility/adding-plug-ins-in-workspace-devfile.png"]
-
-+
-Plug-ins are added in YAML format as array elements in the `components` section.
-. To add plug-ins, append this lines to the `components` section. A record has the following format:
+. To add plug-ins, append your plugin component to the `components` section. A record has the following format:
 +
 ----
+components:
   - type: chePlugin
     reference: >-
-      <registry-base-uri>/v3/plugins/<plugins-group>/<plugin-name>/<version>
+      <registry-base-uri>/v3/plugins/<plugin-publisher>/<plugin-name>/<version>
 ----
 +
 Reference URL examples:
 +
 * pass:[https://custom-registry/v3/plugins/custom-plugin-name/0.0.1/]
 * pass:[https://raw.githubusercontent.com/username/che-custom-plugins/master/v3/plugins/custom-plugin-name/0.0.1/meta.yaml]
 
+[id="adding-a-plug-in-by-shareable-devfile_{context}"]
+== By preparing shareable Devfile
 
-[id="adding-a-plug-in-at-runtime_{context}"]
-== Adding a plug-in at runtime
+This approach brings an advantage of other users being able to contribute to the development, provided they have write access to the repository with the plug-in.
 
-This method adds a plug-in to an already running workspace. The plug-in is not added permanently; it is removed when the workspace stops. This is a good method to test a plug-in without affecting the permanent workspace configuration.
+. Create a devfile with the plug-in development environment and share it with reviewers to provide them with the same environment. Reviewers then must build and run the plug-in to test it, using a script or a set of instructions on how to include the build in the plug-in repository.
 
-. Run the `Deploy Plugin by Id` command, and specify the ID of the plug-in that you want to add to the workspace. A plug-in ID can be:
+. Share with reviwers factory link with a URL to a repository with devfile or a direct URL to content of the devfile.
 +
-** A local directory with plug-ins. All the plug-in binaries (`.theia` files) in the specified directory would be loaded. Example: `local-dir:///home/theia/plugins/`.
-
-** VS Code extension link. To obtain the link, go to the VS Code marketplace, find the needed extension, copy its ID, and append the ID to the `vscode:extension/` prefix.
+Reference URL examples:
 +
-For example, for the link:https://marketplace.visualstudio.com/items?itemName=redhat.vscode-xml[XML language server] plug-in, it is: `vscode:extension/redhat.vscode-xml`.
-
-** Link to plug-in binaries (`.theia` files). Can be both HTTP and HTTPS. Example: `https://domain.net/path/plugin.theia`.
-
-To check if the plug-in loaded successful, open the *Plugins panel* (*View* > *Plugins* in the main menu) and search for the plug-in in the list.
+* {prod-url-secure}/f?url=https://github.com/eclipse/che
+* {prod-url-secure}/f?url=https://raw.githubusercontent.com/eclipse/che/master/devfile.yaml
```

Comment:
```suggestion
.`chePlugin` reference URL pointing to a repository containing a devfile
====
{prod-url}/f?url=https://github.com/eclipse/che
====
+
.`chePlugin` reference URL pointing to a devfile
====
{prod-url}/f?url=https://raw.githubusercontent.com/eclipse/che/master/devfile.yaml
====
```

Reply:
That's not chePlugin reference but a link to Che that will create a workspace via Factory flow with cheplugin inside after plugin author added it into devfile :boom:

Reply:
I've applied

```
.Factory URL pointing to a repository containing a devfile
====
{prod-url}/f?url=https://github.com/eclipse/che
====
+
.Factory URL pointing to a devfile
====
{prod-url}/f?url=https://raw.githubusercontent.com/eclipse/che/master/devfile.yaml
====
```